.github/workflows/nix.yml

```diff
@@ -0,0 +1,23 @@
+name: Nix
+
+on: [push, pull_request]
+jobs:
+  nix:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ['default', 'ghc8102', 'ghc884', 'ghc865']
+        os: [ubuntu-latest, macOS-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cachix/install-nix-action@v12
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
I'd recommend [pinning a commit](https://nix.dev/reference/pinning-nixpkgs.html#possible-url-values) here, otherwise you'll possibly see non-deterministic breakages. Rarely, but not worth the hassle.